sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetReadSupport.scala

```diff
@@ -268,13 +292,23 @@ private[parquet] object ParquetReadSupport {
    * @return A list of clipped [[GroupType]] fields, which can be empty.
    */
   private def clipParquetGroupFields(
-      parquetRecord: GroupType, structType: StructType): Seq[Type] = {
-    val parquetFieldMap = parquetRecord.getFields.asScala.map(f => f.getName -> f).toMap
+      parquetRecord: GroupType,
+      structType: StructType,
+      caseInsensitive: Boolean): Seq[Type] = {
+    val parquetFieldMap = {
+      val pairs = parquetRecord.getFields.asScala.map(f => f.getName -> f)
+      implicit val ordering = if (caseInsensitive) {
+        Ordering.by[String, String](_.toLowerCase)
+      } else {
+        Ordering.String
+      }
+      TreeMap(pairs: _*)
```

Comment:
In the current implementation, we exploit the sorted map keyspace to define an ordering where, if configured, lookups are case-insensitive due to sorting by the downcased key.

We only use the values in the map at one point though, so I'll just roll it back to using a standard Map and add an if/else for downcasing the key when creating/accessing the map.